toolkit/tools/internal/azureblobstorage/azureblobstorage.go

```diff
@@ -136,7 +187,20 @@ func Create(tenantId string, userName string, password string, storageAccount st
 
 	} else if authenticationType == ManagedIdentityAccess {
 
-		credential, err := azidentity.NewDefaultAzureCredential(nil)
+		var credential azcore.TokenCredential
+
+		if azureClientID == "" {
+			logger.Log.Infof("Using DefaultAzureCredential for managed identity access")
+			credential, err = azidentity.NewDefaultAzureCredential(nil)
```

Comment:
Usage of `NewDefaultAzureCredential` is forbidden according to latest S360s we've been getting (see: [this PR](https://github.com/microsoft/azurelinux/pull/14477) for instance). Why do we even need to pass a specific client ID, though? Would `azidentity.NewAzureCLICredential(nil)` combined with logging into Azure before running the toolkit work? It's what we do across our pipelines and seems to be even the case for AZ CLI commands.

Reply:
Decision to use Azure CLI credentials and to map the credential into chroot environment. Based this on top of new 3.0-dev tip which enables use of NewAzureCLICredential call